src/main/java/org/intellij/sdk/codesync/CodeSyncSetup.java

```diff
@@ -0,0 +1,499 @@
+        configRepo.id = repoId;
+        configRepo.token = accessToken;
+        configRepo.email = userEmail;
+
+        try {
+            configRepo.updateRepoBranch(branchName, configRepoBranch);
+            configFile.publishRepoUpdate(configRepo);
+        } catch (InvalidConfigFileError error) {
+            CodeSyncLogger.logEvent(
+                    String.format("[INTELLI_REPO_INIT_ERROR]: Could not update config file. Error %s", error.getMessage())
+            );
+        }
+    }
+
+    public static void uploadToS3(
+            String repoPath, String branchName, String accessToken, String userEmail, Integer repoId,
+            Map<String, Object> fileUrls
+    ) {
+        String originalsRepoBranchPath = Paths.get(ORIGINALS_REPO, repoPath, branchName).toString();
+
+        CodeSyncClient codeSyncClient = new CodeSyncClient();
+
+        for (Map.Entry<String, Object> fileUrl : fileUrls.entrySet()) {
+            File originalsFile = new File(Paths.get(originalsRepoBranchPath, fileUrl.getKey()).toString());
+            if (fileUrl.getValue() == "") {
+                // Skip if file is empty.
+                 continue;
+            }
+            try {
+                codeSyncClient.uploadToS3(
+                        originalsFile,
+                        (Map<String, Object>) fileUrl.getValue()
+                );
+            } catch (RequestError | ClassCastException error) {
+                CodeSyncLogger.logEvent(
+                        String.format("[INTELLI_REPO_INIT_ERROR]: Could not upload file to S3. Error %s", error.getMessage())
+                );
+            }
+        }
+    }
+
+    /*
+    List absolute file paths of all the files in a directory.
+
+    This is recursively list all the files containing in the given directory and all its sub-directories.
+     */
+    public static String[] listFiles(String directory) {
+        String[] filePaths = {};
+        try {
+            filePaths = Files.walk(Paths.get(directory))
+                    .filter(Files::isRegularFile)
+                    .map(name -> name.toString().replace(directory, ""))
+                    .map(name -> name.replaceFirst("/", ""))
+                    .filter((name) -> !Utils.shouldIgnoreFile(name, directory))
+                    .map((name) -> String.format("%s/%s", directory.replaceFirst("/$",""), name))
+                    .toArray(String[]::new);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return filePaths;
+    }
+
+    public static void createSyncIgnore(String repoPath) {
+        String syncIgnorePath = String.format("%s/.syncignore", repoPath);
+        String gitIgnorePath = String.format("%s/.gitignore", repoPath);
+        File syncIgnoreFile = new File(syncIgnorePath);
+        File gitIgnoreFile = new File(gitIgnorePath);
+
+        // no need to create .syncignore if it already exists.
+        if (syncIgnoreFile.exists()) {
+            return ;
```

Comment:
I think in this case, we still need user input to proceed further, user might want to edit the syncignore or continue to next step